homeassistant/components/remootio/utils.py

```diff
@@ -0,0 +1,153 @@
+"""Utility methods for the Remootio integration."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from logging import Logger
+
+from aioremootio import (
+    ConnectionOptions,
+    LoggerConfiguration,
+    RemootioClient,
+    RemootioClientAuthenticationError,
+)
+from aioremootio.enums import State
+
+from homeassistant import core
+from homeassistant.exceptions import (
+    ConfigEntryAuthFailed,
+    ConfigEntryError,
+    ConfigEntryNotReady,
+)
+from homeassistant.helpers import aiohttp_client
+
+from .const import EXPECTED_MINIMUM_API_VERSION, REMOOTIO_DELAY, REMOOTIO_TIMEOUT
+from .exceptions import (
+    UnsupportedRemootioApiVersionError,
+    UnsupportedRemootioDeviceError,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _wait_for_connected(remootio_client: RemootioClient) -> bool:
+    while not remootio_client.connected:
+        await asyncio.sleep(REMOOTIO_DELAY)
+
+    return remootio_client.connected
```

Comment:
@elupus is correct, the return value of this function is useless and just causes unwanted nesting when called.

```suggestion
async def _wait_for_connected(remootio_client: RemootioClient) -> None:
    while not remootio_client.connected:
        await asyncio.sleep(REMOOTIO_DELAY)
```